pac4j-oauth/src/main/java/org/pac4j/oauth/client/GenericOAuth20Client.java

```diff
@@ -106,12 +110,22 @@ protected GenericOAuth20Profile extractUserProfile(String body) {
         if (attributesDefinition != null) {
             profile.setAttributesDefinition(attributesDefinition);
         }
-        final JsonNode json = JsonHelper.getFirstNode(body);
+        final JsonNode json = JsonHelper.getFirstNode(body, getProfileNodePath());
         if (json != null) {
             profile.setId(JsonHelper.getElement(json, "id"));
             for (final String attribute : profile.getAttributesDefinition().getPrimaryAttributes()) {
                 profile.addAttribute(attribute, JsonHelper.getElement(json, attribute));
             }
+            for (final String attribute : profile.getAttributesDefinition().getSecondaryAttributes()) {
```

Comment:
This raises an interesting about the way we find attributes, besides the Scribe upgrade. I think you lean towards path matching without going that far completely.